internal/mode/static/nginx/config/maps.go

```diff
@@ -21,7 +22,67 @@ func executeMaps(conf dataplane.Configuration) []executeResult {
 	return []executeResult{result}
 }
 
-func buildAddHeaderMaps(servers []dataplane.VirtualServer) []http.Map {
+func executeStreamMaps(conf dataplane.Configuration) []executeResult {
+	maps := createStreamMaps(conf)
+
+	result := executeResult{
+		dest: streamConfigFile,
+		data: helpers.MustExecuteTemplate(mapsTemplate, maps),
+	}
+
+	return []executeResult{result}
+}
+
+func createStreamMaps(conf dataplane.Configuration) []shared.Map {
+	maps := make([]shared.Map, 0)
+	portsToMap := make(map[int32]int)
```

Comment:
talked offline; @sarthyparty is going to refactor this to be a `map[port]shared.Map` and then return a list of the shared.Map.